numpy/core/src/multiarray/experimental_public_dtype_api.c

```diff
@@ -196,11 +197,17 @@ PyArrayInitDTypeMeta_FromSpec(
          * It is up to the user to get this right, and slots are sorted
          * exactly like they are stored right now:
          */
-        void **current = (void **)(&(
-                NPY_DT_SLOTS(DType)->discover_descr_from_pyobject));
-        current += slot - 1;
-        *current = pfunc;
+        if (slot <= 8) {
+            // slot > 8 are PyArray_ArrFuncs
```

Comment:
I wonder if we should use a larger ID to allow easier inserting more funcs before them, then we can keep up the "its sorted" thing for longer, which might be nice.

Reply:
Yep, good idea - done.